sw/host/opentitanlib/src/transport/common/uart.rs

```diff
@@ -245,23 +245,49 @@ impl SerialPortExclusiveLock {
                 }
             }
         }
-        let mut lockfile = OpenOptions::new()
+        match OpenOptions::new()
             .write(true)
             .create_new(true)
-            .open(&lockfilename)?;
-        if PID_FILE_LEN != lockfile.write(format!("{:10}\n", nix::unistd::getpid()).as_bytes())? {
-            bail!(TransportError::OpenError(
-                port_name.to_string(),
-                "Error writing lockfile".to_string()
-            ));
+            .open(&lockfilename)
+        {
+            Ok(mut lockfile) => {
+                if PID_FILE_LEN
+                    != lockfile.write(format!("{:10}\n", nix::unistd::getpid()).as_bytes())?
+                {
+                    bail!(TransportError::OpenError(
+                        port_name.to_string(),
+                        "Error writing lockfile".to_string()
+                    ));
+                }
+                Ok(Self {
+                    lockfilename: Some(lockfilename),
+                })
+            }
+            Err(ref e) if e.kind() == ErrorKind::AlreadyExists => {
+                // Possibly some other console created this file in between us unsuccessfuly
+                // attempting to open it, and then attempting to create it.
+                bail!(TransportError::OpenError(
+                    port_name.to_string(),
+                    "Device is locked".to_string()
+                ))
+            }
+            Err(ref e)
+                if e.kind() == ErrorKind::NotFound || e.kind() == ErrorKind::PermissionDenied =>
+            {
+                // /var/lock may not exist or not allow users to create files.  We proceed without
+                // locking here.  (We will still use flock() in `SerialPortUart::open()`.)
+                Ok(Self { lockfilename: None })
+            }
+            Err(e) => bail!(e),
```

Comment:
I find `std::io::Error` messages annoying because they can lack context. Having said that, it is nice knowing why it has failed to write the lock file, so long as you know it is the lock file it is attempting to write.

Maybe what was originally used would be better for now?

```suggestion
            Err(e) => bail!(TransportError::OpenError(
                port_name.to_string(),
                "Error writing lockfile".to_string()
            )),
```

Reply:
Originally, there was just the operator `?` after both the `open()` and the `write()` call.  I do see that propagating context (that we are operating on a lock file, and a particular path), as well as the error code, will be useful.  I have enhanced several error paths to pass all the relevant information in a string.